libraries/botbuilder-dialogs-adaptive/tests/beginSkill.test.js

```diff
@@ -126,6 +128,18 @@ describe('BeginSkill', function () {
 
         await adapter.send('test');
     });
+    
+    it('should respect allow interruptions settings', async () => {
+        dialog.allowInterruptions = new BoolExpression(false);
+        const adapter = new TestAdapter(async (context) => {
+            const dc = await dialogs.createContext(context);
+            
+            const bubbling = await dialog.onDialogEvent(dc, { name: DialogEvents.activityReceived });
+            strictEqual(bubbling, true);
+        });
+        
+        await adapter.send('test');
```

Comment:
As far as I can tell this test won't wait until the contents of the `TestAdapter` function are executed. To test this you could add a `strictEqual(true, false)` inside that method to verify. I _think_ you'll want to write this as:

```javascript
if('...', async (done) => {}
  const adapter = new TestAdapter(async context => {
    ...
    done();
  });

  await adapter.send('test');
);
```

Reply:
I put the `strictEqual(true, false)` inside that method and the assertion failed. Actually, the async function what we defined in the constructor of TestAdapter is the logic we will execute when calling `adapter.send()`, so the assertions inside that method will be executed.
BTW, using `done()` is not allowed because async functions will always return promises and I got the following error:
```
Error: Resolution method is overspecified. Specify a callback *or* return a Promise; not both.
```

Reply:
Ah - makes sense. I wasn't sure if the logic executed by `adapter.send` was itself executed asynchronously.